api/lock_api.go

```diff
@@ -144,15 +142,8 @@ type Committer struct {
 	Email string `json:"email"`
 }
 
-// CurrentCommitter returns a Committer instance populated with the same
-// credentials as would be used to author a commit. In particular, the
-// "user.name" and "user.email" configuration values are used from the
-// config.Config singleton.
-func CurrentCommitter() Committer {
```

Comment:
Hmm. While I like the `Unmarshal` approach to pure data structure population it feels a bit too "magic" and implicit to be using directly on objects like `Client`. Accidental naming clashes could cause all sorts of hard to trace issues. While I agree that we should move away from accessing the global config everywhere, passing an explicit configuration structure to other packages seems a lot easier to understand & trace.

However, in order to remove the dependency on a "fat" `config.Configuration` I'd be totally happy to have separate package-specific configuration structures, which *themselves* are populated using `Unmarshal`, because that's a much more straight forward data encapsulation case, rather than having a more general purpose struct like `Client` being populated directly. I'm concerned that because `Client` is general purpose you could add members to it for other reasons and accidentally cause a name clash which bleeds into the config exchange. That won't happen if the package-specific config structure is only ever used for config fields.

The problem with doing this right now is that to do it on `locking` I'd also have to do it on `api` since that still needs the full config, which in turn would have to do it on `httputil` and `auth` and I'm not sure if that's even the end of it 😱 It needs doing but can't be done in isolation in `locking`, which makes me think it should be a separate refactor.